boa/src/syntax/parser/tests.rs

```diff
@@ -34,10 +40,16 @@ where
 /// Checks that the given javascript string creates a parse error.
 // TODO: #[track_caller]: https://github.com/rust-lang/rust/issues/47809
 pub(super) fn check_invalid(js: &str) {
-    let mut lexer = Lexer::new(js);
-    lexer.lex().expect("failed to lex");
+    let mut lexer = Lexer::new(js.as_bytes());
+    // lexer.lex().expect("failed to lex");
+
+    let mut tokens = Vec::new();
+
+    for token in lexer {
+        tokens.push(token.expect("failed to lex"));
+    }
```

Comment:
This can be improved further:
```suggestion
    let tokens = lexer.collect::<Result<Vec<_>, _>().expect("failed to lex");
```